modules/ROOT/pages/type-definitions/directives/custom-logic.adoc

```diff
@@ -0,0 +1,641 @@
+
+[source, graphql, indent=0]
+----
+type Query {
+    name(value: String): String @cypher(statement: "RETURN $value AS res", columnName: "res")
+}
+----
+
+The following GraphQL query returns the parameter `value`:
+
+[source, graphql, indent=0]
+----
+query {
+  name(value: "Jane Smith")
+}
+----
+
+
+=== Usage examples
+
+The `@cypher` directive can be used in different contexts, such as the ones described in this section.
+
+[[type-definitions-cypher-object-usage]]
+==== On an object type field
+
+In the example below, a field `similarMovies` is bound to the `Movie` type, to find other movies with an overlap of actors:
+
+[source, graphql, indent=0]
+----
+type Actor {
+    actorId: ID!
+    name: String
+    movies: [Movie!]! @relationship(type: "ACTED_IN", direction: OUT)
+}
+
+type Movie {
+    movieId: ID!
+    title: String
+    description: String
+    year: Int
+    actors(limit: Int = 10): [Actor!]!
+        @relationship(type: "ACTED_IN", direction: IN)
+    similarMovies(limit: Int = 10): [Movie]
+        @cypher(
+            statement: """
+            MATCH (this)<-[:ACTED_IN]-(:Actor)-[:ACTED_IN]->(rec:Movie)
+            WITH rec, COUNT(*) AS score ORDER BY score DESC
+            RETURN rec LIMIT $limit
+            """,
+            columnName: "rec"
+        )
+}
+----
+
+==== On a query type field
+
+The following example demonstrates a query to return all of the actors in the database:
+
+[source, graphql, indent=0]
+----
+type Actor {
+    actorId: ID!
+    name: String
+}
+
+type Query {
+    allActors: [Actor]
+        @cypher(
+            statement: """
+            MATCH (a:Actor)
+            RETURN a
+            """,
+            columnName: "a"
+        )
+}
+----
+
+==== On a mutation type field
+
+The following example demonstrates a mutation using a Cypher query to insert a single actor with the specified name argument:
+
+[source, graphql, indent=0]
+----
+type Actor {
+    actorId: ID!
+    name: String
+}
+
+type Mutation {
+    createActor(name: String!): Actor
+        @cypher(
+            statement: """
+            CREATE (a:Actor {name: $name})
+            RETURN a
+            """,
+            columnName: "a"
+        )
+}
+----
+
+
+== `@coalesce`
+
+When translating from GraphQL to Cypher, any instances of fields to which this directive is applied will be wrapped in a `coalesce()` function in the WHERE clause (see https://neo4j.com/developer/kb/understanding-non-existent-properties-and-null-values/#_use_coalesce_to_use_a_default_for_a_null_value).
+
+This directive helps querying against non-existent properties in a database. 
+However, it is encouraged to populate these properties with meaningful values if this is becoming the norm. The `@coalesce`directive is a primitive implementation of the function which only takes a static default value as opposed to using another property in a node or a Cypher expression.
```

Comment:
This could perhaps become a note:

```suggestion
However, it is encouraged to populate these properties with meaningful values if it becomes the norm. 
The `@coalesce`directive is a primitive implementation of the function which only takes a static default value as opposed to using another property in a node or a Cypher expression.
```